docs/components/modeler/desktop-modeler/element-templates/defining-templates.md

```diff
@@ -73,6 +77,40 @@ For example, given the following `$schema` definition, the application takes `0.
 
 The JSON schema versioning is backward-compatible, meaning that all versions including or below the current one are supported.
 
+## Execution platform compatibility
+
+You can define template compatibility with different execution platforms using the `engines` property.
+
+This property is a dictionary object, where the execution platform names are the keys, and the [semantic version](https://semver.org/) ranges are the values.
+
+For example, the following `engines` definition specifies that the template is compatible with Camunda 8.6 or higher.
+
+```json
+"engines": {
+  "camunda": ">8.5"
+}
+```
+
+Compatibility is only validated if the platform version is provided by both the template and the modeler. In the example below, the template is compatible with specific versions of both Desktop and Web Modeler, but it requires Camunda version 8.6 or higher for both:
+
+```json
+"engines": {
+  "camunda": ">8.5",
+  "camundaDesktopModeler": "5.30",
+  "camundaWebModeler": "8.6.5"
```

Comment:
I'd rather not use an example value which leads to trouble very quickly. Pinned compatibility is unusual. Let's use values which can be potentially used, e.g. `>=5.30`, `^8.5.5`.